packages/ubuntu_desktop_installer/lib/utils/version_extractor.dart

```diff
@@ -0,0 +1,65 @@
+import 'dart:io';
+
+class VersionExtractor {
+  static VersionExtractor? _instance;
+
+  static VersionExtractor get instance {
+    if (_instance == null) {
+      _instance = VersionExtractor._();
+    }
+    return _instance!;
+  }
+
+  VersionExtractor._();
+
+  String loadedVersion = '';
+
+  String _dummy() {
+    return 'Ubuntu';
+  }
+
+  String extractLocalVersion(String content) {
+    return content.substring(content.indexOf("\"") + 1, content.length - 1);
```

Comment:
This method doesn't need to be public, I think.
And the parsing makes unnecessary assumptions (that the closing double quote is the last character on the line). You could use `String.lastIndexOf` or `String.split`.

Reply:
@oSoMoN I made those method public because I wanted to test just them. If we want to keep the test, it might be good to extract parsing logic to serparate class, what would you prefer?

Reply:
That's not the correct approach: tests should exercise (thoroughly) public methods because they are public, the rest is implementation details. In fact I'd argue that these methods shouldn't even be separate methods, since they're used in only one place their implementation should be inlined.

Reply:
I agree, still it would be nice to have test for those parsers. Seems like file package ( https://stackoverflow.com/questions/61504873/testing-mocking-file-io-in-flutter ) allows for in memory file mocking. Should we move to this approach?

Reply:
Yes, we could do that. If I understand correctly, that would require passing a `FileSystem` instance to the `VersionExtractor` constructor, which exposes an implementation detail, but that's an acceptable trade-off to achieve good test coverage I suppose.

Reply:
Sounds good, I will play with it and report my findings or just implement that if it's straighforward :+1: